### PR TITLE
enhance(pods): various airtable pod enhancements

### PR DESCRIPTION
**Pull request by kevinslin**
- chore(internal): clean up unused bootstrap scripts
- error message when single tag field is wrong
- introduce new note props type
- - make onExportComplete be async - await on async exports inside BaseExportPod
- update export pod logic

# Pull Request Checklist

If this is your first time submitting a pull request to Dendron, copy and paste the full [Dendron Review Checklist](https://docs.dendron.so/notes/1EoNIXzgmhgagqcAo9nDn.html) into this request and check off each item as necessary. 

This template contains the short checklist which is used by the Dendron core team. 

## Testing
- [ ] [Write Tests](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#writing-tests) 
- [ ] [Confirm existing tests pass](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#executing-tests)
- [ ] [Confirm manual testing](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#manual-testing) 
- [ ] If your tests changes an existing snapshot, make sure that snapshots are [updated](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#updating-test-snapshots)
- [ ] If you are adding a new language feature (graphically visible in VS Code/preview/publishing), make sure that it is included in the [test workspace](https://docs.dendron.so/notes/dtMsF12SF2SUhLN10sYe2.html)
### Docs
- [ ] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
### Analytics
- [ ] if you are adding analytics related changes, make sure the [Telemetry](https://docs.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### CODE REVIEW

File: packages/plugin-core/src/commands/pods/BaseExportPodCommand.ts [hunk 1]
```diff
@@ -179,21 +180,16 @@ export abstract class BaseExportPodCommand<
             break;
           }
           case PodExportScope.Note: {
-            const promises = [];
```
Comment (jonathanyeung) [inline packages/plugin-core/src/commands/pods/BaseExportPodCommand.ts:182]: Removing this pattern leads to an ESLint warning about having await inside a for loop.

File: packages/pods-core/src/builtin/AirtablePod.ts [hunk 2]
```diff
@@ -178,6 +179,11 @@ export class AirtableUtils {
     return _.flatten(out);
   }
 
+  /**
+   * Maps a {@linkk SrcFieldMappingV2["type"]} to a field on {@link NoteProps}
```
Comment (jonathanyeung) [inline packages/pods-core/src/builtin/AirtablePod.ts:183]: nit: linkk

File: packages/pods-core/src/builtin/AirtablePod.ts [hunk 3]
```diff
@@ -390,7 +398,7 @@ export class AirtableUtils {
       records.map(async (ent) => {
         const airtableId = ent.id;
         const dendronId = ent.fields["DendronId"] as string;
-        const note = engine.notes[dendronId];
+        const note = engine.notes[dendronId] as NotePropsWithOptionalCustom;
```
Comment (jonathanyeung) [inline packages/pods-core/src/builtin/AirtablePod.ts:401]: Is this actually necessary? what if we do something akin to `isStringNullOrEmpty()` on the NoteAirtableId below.  Can we avoid a cast and a new type definition?

**Review by jonathanyeung — CHANGES_REQUESTED** [review]

File: packages/plugin-core/src/commands/pods/BaseExportPodCommand.ts (hunk 1 above)
Comment (kevinslin) [reply packages/plugin-core/src/commands/pods/BaseExportPodCommand.ts:182]: that's why I added a es-lint ignore  at the top of the file
`/* eslint-disable no-await-in-loop */`

File: packages/pods-core/src/builtin/AirtablePod.ts (hunk 2 above)
Comment (kevinslin) [reply packages/pods-core/src/builtin/AirtablePod.ts:183]: fixed

File: packages/pods-core/src/builtin/AirtablePod.ts (hunk 3 above)
Comment (kevinslin) [reply packages/pods-core/src/builtin/AirtablePod.ts:401]: our actual `NoteProps` definition is wrong. `custom` is optional and can be undefined. we actually need to refactor the definition (or the code) but that was a much bigger change, hence this type as a stopgap.

we'll fix this as part of january core feature maturity milestones

File: packages/plugin-core/src/commands/pods/BaseExportPodCommand.ts (hunk 1 above)
Comment (jonathanyeung) [reply packages/plugin-core/src/commands/pods/BaseExportPodCommand.ts:204]: But why change it if the current pattern doesn't warn?

**Review by jonathanyeung — APPROVED** [review]

**Pull request merged**